lib/fuse.c

```diff
@@ -3566,7 +3566,7 @@ static int fill_dir_plus(void *dh_, const char *name, const struct stat *statp,
 		return 1;
 	}
 
-	if (statp && (flags & FUSE_FILL_DIR_PLUS)) {
+	if (off && statp && (flags & FUSE_FILL_DIR_PLUS)) {
```

Comment:
Not sure I understand this change. Why do we want to throw away most of the `statp` information when seeking is not supported?

Reply:
Oh, I did not want to deprive any project of the stat information when available. This intent was not mentioned in the rationale which led to removing the "off == 0" check.

Rather, I think this check was there for a reason (possibly because some file system needed it), and I just wanted to avoid a possible regression.

Now, if the stat information is a useful feature on the first returned entry in readdir_plus mode (usually this is "."), this is a good rationale for removing the check, so let me keep it removed. This has no effect in my use cases.

Reply:
Got you. Yes, I'd rather leave this as-is.

Reply:
This proposal keeps the off == 0 test as is and only deals with the non-plus readdir(3). Also the passthrough check has been extended to test readdir(3) in both modes.